dotnet/src/Functions/Functions.Yaml/PromptExecutionSettingsTypeConverter.cs

```diff
@@ -0,0 +1,105 @@
+            var propertyName = parser.Consume<Scalar>().Value;
+            switch (propertyName)
+            {
+                case "model_id":
+                    executionSettings.ModelId = s_deserializer.Deserialize<string>(parser);
+                    break;
+                case "function_choice_behavior":
+#pragma warning disable SKEXP0001
+                    executionSettings.FunctionChoiceBehavior = s_deserializer.Deserialize<FunctionChoiceBehavior>(parser);
+#pragma warning restore SKEXP0001
+                    break;
+                default:
+                    (executionSettings.ExtensionData ??= new Dictionary<string, object>()).Add(propertyName, s_deserializer.Deserialize<object>(parser));
+                    break;
+            }
+        }
+        parser.MoveNext(); // Move past the MappingEnd event  
+        return executionSettings;
+    }
+
+    /// <inheritdoc/>
+    public void WriteYaml(IEmitter emitter, object? value, Type type)
+    {
+        throw new NotImplementedException();
+    }
+
+    /// <summary>
+    /// Creates and register a <see cref="TypeDiscriminatingNodeDeserializer" /> for polymorphic deserialization of <see cref="FunctionChoiceBehavior" />.
+    /// </summary>
+    /// <param name="options">The <see cref="ITypeDiscriminatingNodeDeserializerOptions" /> to configure the <see cref="TypeDiscriminatingNodeDeserializer" />.</param>
+    private static void CreateAndRegisterTypeDiscriminatingNodeDeserializer(ITypeDiscriminatingNodeDeserializerOptions options)
+    {
+#pragma warning disable SKEXP0001
+        var attributes = typeof(FunctionChoiceBehavior).GetCustomAttributes(false);
+
+        // Getting the type discriminator property name - "type" from the JsonPolymorphicAttribute.
+        var discriminatorKey = attributes.OfType<JsonPolymorphicAttribute>().Single().TypeDiscriminatorPropertyName;
+        if (string.IsNullOrEmpty(discriminatorKey))
+        {
+            throw new InvalidOperationException("Type discriminator property name is not specified.");
+        }
+
+        var discriminatorTypeMapping = new Dictionary<string, Type>();
+
+        // Getting FunctionChoiceBehavior subtypes and their type discriminators registered for polymorphic deserialization.
+        var derivedTypeAttributes = attributes.OfType<JsonDerivedTypeAttribute>();
+        foreach (var derivedTypeAttribute in derivedTypeAttributes)
+        {
+            var discriminator = derivedTypeAttribute.TypeDiscriminator?.ToString();
+            if (string.IsNullOrEmpty(discriminator))
+            {
+                throw new InvalidOperationException($"Type discriminator is not specified for the {derivedTypeAttribute.DerivedType} type.");
+            }
+
+            discriminatorTypeMapping.Add(discriminator!, derivedTypeAttribute.DerivedType);
+        }
+
+        options.AddKeyValueTypeDiscriminator<FunctionChoiceBehavior>(discriminatorKey!, discriminatorTypeMapping);
+#pragma warning restore SKEXP0010
```

Comment:
Different closing pragma (recommend removing all of them) as previous comment.
```suggestion
#pragma warning restore SKEXP0010
```

Reply:
Thanks, fixed.